File: hub/src/util/color_transition.py
from light.rgbw import RGBW

class ColorTransition:
    @staticmethod
    def linear(tNorm, srcRgb, dstRgb):
        r = (1 - tNorm) * srcRgb.r + tNorm * dstRgb.r
        g = (1 - tNorm) * srcRgb.g + tNorm * dstRgb.g
        b = (1 - tNorm) * srcRgb.b + tNorm * dstRgb.b
        return RGBW(int(r), int(g), int(b), 0)

File: hub/drumlite-js/src/util/ColorTransition.ts
import RGB from "../light/RGB"

export default class ColorTransition {
    public static linear(tNorm: number, srcRgb: RGB, dstRgb: RGB) {
        const r = (1 - tNorm) * srcRgb.r + tNorm * dstRgb.r;
        const g = (1 - tNorm) * srcRgb.g + tNorm * dstRgb.g;
        const b = (1 - tNorm) * srcRgb.b + tNorm * dstRgb.b;
        return new RGB(Math.round(r), Math.round(g), Math.round(b));
    }
}
